bsp4j/src/main/java/ch/epfl/scala/bsp4j/CppExtension.xtend

```diff
@@ -0,0 +1,43 @@
+package ch.epfl.scala.bsp4j
+
+import java.util.List
+import org.eclipse.lsp4j.jsonrpc.validation.NonNull
+import org.eclipse.lsp4j.generator.JsonRpcData
+
+@JsonRpcData
+class CppBuildTarget {
+  @NonNull CppPlatform platform
```

Comment:
Should this be an enum or a string? A string is what is used for the java version, but that might be because that usually uses semantic versioning, whilst in c++ we have to handle cases such as `C++11`

Reply:
Probably just a String, so that implementor-side extensions are easier and don't later conflict with the mainline protocol. We can have a list of predefined constants with agreed on meaning in the protocol.